### PR TITLE
Adds kube state metrics component

### PR DESCRIPTION
**Pull request by JosephSalisbury**
Towards https://github.com/giantswarm/giantswarm/issues/1482

### CODE REVIEW

File: tls_assets.go [hunk 1]
```diff
@@ -76,6 +78,13 @@ var ClusterComponents = []ClusterComponent{
 	ServiceAccountComponent,
 }
 
+// MonitoringComponents is a slice enumerating all the components that make up
+// monitoring.
+var MonitoringComponents = []ClusterComponent{
```
Comment (JosephSalisbury) [inline tls_assets.go:83]: I don't want to add monitoring components to `ClusterComponents`, as they're not required for the guest cluster - these are for monitoring systems to access the guest cluster. I do want a list of them though, so we can clean them up properly.

This also avoids issues with `SearchCerts` for older clusters that don't have certificates for guest cluster scraping.

File: tls_assets.go [hunk 2]
```diff
@@ -76,6 +78,13 @@ var ClusterComponents = []ClusterComponent{
...
+	PrometheusComponent,
```
Comment (JosephSalisbury) [inline tls_assets.go:84]: This is probably named badly at this point. We should be using separate certificates for each monitoring components - this is for the API server. Learning as we go along. Planning on cleaning this up shortly.

**Review by MarcelMue — APPROVED** [review]
LGTM, the new slice for monitoring components makes sense to me.
Let's iterate on the naming.

File: tls_assets.go (hunk 2 above)
Comment (xh3b4sd) [reply tls_assets.go:84]: Works for me. Maybe just add some comments where you can explain things.

**Review by xh3b4sd — APPROVED** [review]

**Pull request merged**